### PR TITLE
Quinte dev branch

### PR DESCRIPTION
**Pull request by Quilver**
If and While statements

### CODE REVIEW

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 1]
```diff
@@ -238,5 +329,89 @@ private void ExpressionQuery(int level)
...
+		}
+		catch (Exception e)
+		{
+			_logger.Warning(e.Message);
+		}
+	}
+	private void ElseStatements(int level) {
+		if (Match(TokenType.String))
+		{
+			if (CheckElse())
```
Comment (A-Russell23) [inline C_Flat_Interpreter/Parser/Parser.cs:366]: `CheckElse` could be added to line 364, like you've done on line 337

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 2]
```diff
@@ -238,5 +329,89 @@ private void ExpressionQuery(int level)
...
+				Advance(level + 1);
+			SubBlock(level + 1);
```
Comment (A-Russell23) [inline C_Flat_Interpreter/Parser/Parser.cs:368]: Ensure `subblock` is only called if `CheckElse` returns true

**Review by A-Russell23 — CHANGES_REQUESTED** [review]
This is a good start. I've left a few comments about some things that need a look over. If you could sort them out that would be great. If you don't agree or you're not sure what I mean, I'm happy to have a discussion with you and/or the team.

A couple notes...

Just for future reference, its great that you've implemented while loops but it would have been better for you to finish up all of your assigned work (such as unit tests) before branching out to other things. If you had put that extra time into getting tests to work, this might have been ready to merge on Monday. I'd say its always better to have a working project with less functionality rather than a broken project with more functionality. Its also worth noting that the longer it takes to get stuff merged, the more conflicts are likely to happen once we are able to merge this work, so this should always be a priority. Don't forget to think about how your actions might affect the wider team and the project as a whole :)

Another quick note, for `ParserUnit.cs` you'll need to add tests for the rest of the code, not just for the if/else statements you've added. You also need to include tests for errors.

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 3]
```diff
@@ -73,39 +103,98 @@ private void Advance(int level)
 		_tokenType = _tokens[_currentIndex].Type;
 		_logger.Information("advance() called at level {level}. Next token is {@token}", level, _tokens[_currentIndex]);
 	}
+	private void PrintCurrentLevel()
```
[the remark this answers is not in the document]
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs]: Removed

File: C_Flat_Interpreter/Parser/Parser.cs (hunk 2 above)
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs:351]: Got it

File: C_Flat_Interpreter/Parser/Parser.cs (hunk 1 above)
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs:349]: Got it

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 4]
```diff
@@ -61,10 +84,17 @@ private bool Match(TokenType tokenType)
 		}
 		return true;
 	}
-
+	Stack<int> tokenIndexStack;
```
[the remark this answers is not in the document]
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs]: Removed

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 5]
```diff
@@ -73,39 +103,98 @@ private void Advance(int level)
...
-		Reset();
+		_currentIndex = 0;
 		Statement(0);
 		if (_currentIndex >= _totalTokens) return 0;
 		_logger.Error("Syntax error! Could not parse Token: {@token}", _tokens[_currentIndex]); //todo - Create test for this!
 		return 1;
 	}
 	
 	//EBNF Functions
+	private void Block(int level)
```
[the remark this answers is not in the document]
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs]: I'll move it down for now, but we should discuss implementation of multiple lines of code

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 6]
```diff
@@ -238,5 +329,89 @@ private void ExpressionQuery(int level)
 		Advance(level + 1);
 		Expression(level + 1);
 	}
+    #endregion
+    #region Conditions and iterators
+	private void IfStatements(int level)
+    {
+		_logger.Information("if-statement() called at level {level}", level);
+		if (Match(TokenType.String) && CheckIf())
```
[the remark this answers is not in the document]
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs:318]: I'm not sure what you mean by this

File: C_Flat_Interpreter/Parser/Parser.cs [hunk 7]
```diff
@@ -238,5 +329,89 @@ private void ExpressionQuery(int level)
...
+		LogicStatement(level + 1);
+		if (Match(TokenType.RightParen)) Advance(level + 1);
+		else
+		{
+			_logger.Error("Syntax Error! Expected \"(\" actual: {@word} ", _tokens[_currentIndex].Word);
+			return;
+		}
+		SubBlock(level + 1);
+	}
+	private void SubBlock(int level) {
```
[the remark this answers is not in the document]
Comment (Quilver) [reply C_Flat_Interpreter/Parser/Parser.cs:380]: I prefer having smaller functions. Although perhaps I had an issue with poorly naming the functions

**Comment by KThompson-Lane** [comment]
Closing as this is was superseded by #14

**Pull request closed without merging**